libcomps/src/python/src/pycomps.c

```diff
@@ -512,9 +511,9 @@ PyObject* PyCOMPS_filter_arches(PyObject *self, PyObject *other) {
         arches = COMPS_OBJECT_CREATE(COMPS_ObjList, NULL);
         for (Py_ssize_t x=0; x < PyList_Size(other); x++) {
             item = PyList_GetItem(other, x);
+            char *str;
             __pycomps_arg_to_char(item, &str);
-            comps_objlist_append_x(arches, (COMPS_Object*)comps_str(str));
-            free(str);
+            comps_objlist_append_x(arches, (COMPS_Object*)comps_str_x(str));
```

Comment:
`__pycomps_arg_to_char(item, &str)` return value should be checked and `NULL` returned in case it is <> 0 - see https://bugzilla.redhat.com/show_bug.cgi?id=1938759#c9 . Otherwise the function ends with a core dump. Simple reproducer:

```
import libcomps
INVALID_UTF8_CHAR = '\udcfd'
c = libcomps.Comps()
c.arch_filter([INVALID_UTF8_CHAR])
```

Reply:
I see, I have added a new commit for that.
Thank you.